sdk/storage/storage-blob/rollup.base.config.js

```diff
@@ -0,0 +1,185 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import nodeResolve from "rollup-plugin-node-resolve";
+import multiEntry from "rollup-plugin-multi-entry";
+import cjs from "rollup-plugin-commonjs";
+import json from "rollup-plugin-json";
+import replace from "rollup-plugin-replace";
+import { uglify } from "rollup-plugin-uglify";
+import sourcemaps from "rollup-plugin-sourcemaps";
+import shim from "rollup-plugin-shim";
+// import visualizer from "rollup-plugin-visualizer";
+
+import path from "path";
+
+const version = require("./package.json").version;
+const banner = [
+  "/*!",
+  ` * Azure Storage SDK for JavaScript - Blob, ${version}`,
+  " * Copyright (c) Microsoft and contributors. All rights reserved.",
+  " */"
+].join("\n");
+
+const pkg = require("./package.json");
+const depNames = Object.keys(pkg.dependencies);
+const production = process.env.NODE_ENV === "production";
+
+export function nodeConfig(test = false) {
+  const externalNodeBuiltins = [
+    "@azure/ms-rest-js",
+    "crypto",
+    "fs",
+    "events",
+    "os",
+    "stream"
+  ];
+  const baseConfig = {
+    input: "dist-esm/src/index.js",
+    external: depNames.concat(externalNodeBuiltins),
+    output: {
+      file: "dist/index.js",
+      format: "cjs",
+      sourcemap: true
+    },
+    plugins: [
+      sourcemaps(),
+      replace({
+        delimiters: ["", ""],
+        values: {
+          // replace dynamic checks with if (true) since this is for node only.
+          // Allows rollup's dead code elimination to be more aggressive.
+          "if (isNode)": "if (true)"
+        }
+      }),
+      nodeResolve({ preferBuiltins: true }),
+      cjs(),
+      json()
+    ]
+  };
+
+  if (test) {
+    // entry point is every test file
+    baseConfig.input = [
+      "dist-esm/test/*.spec.js",
+      "dist-esm/test/node/*.spec.js"
+    ];
+    baseConfig.plugins.unshift(multiEntry({ exports: false }));
+
+    // different output file
+    baseConfig.output.file = "dist-test/index.node.js";
+
+    // mark assert as external
+    baseConfig.external.push("assert", "fs", "path");
+
+    baseConfig.onwarn = warning => {
+      if (warning.code === "THIS_IS_UNDEFINED") {
+        // This error happens frequently due to TypeScript emitting `this` at the
+        // top-level of a module. In this case its fine if it gets rewritten to
+        // undefined, so ignore this error.
+        return;
+      }
+
+      if (
```

Comment:
Do you need this custom onwarn machinery? It's included in the template as an example, but if it's not needed here feel free to remove it.

Reply:
I remember facing an issue with "this" which got resolved when I added this(for browser)
I'll verify once again.

Reply:
**Removed chai warning block(not required).
"THIS_IS_UNDEFINED" is required